src/components/editor-page/note-frontmatter/extract-frontmatter.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { allCharsEqual } from '../utils'
+import { FrontmatterExtractionResult } from './types'
+
+/**
+ * Extracts a frontmatter block from a given multiline string.
+ * A valid frontmatter block requires the content to start with a line containing at least three dashes.
+ * The block is terminated by a line containing the same amount of dashes or dots as the first line.
+ * @param content The multiline string from which the frontmatter should be extracted.
+ * @return { frontmatterPresent } false if no frontmatter block could be found, true if a block was found.
+ *         { rawFrontmatterText } if a block was found, this property contains the extracted text without the fencing.
+ *         { frontmatterLines }   if a block was found, this property contains the number of lines to skip from the
+ *                                given multiline string for retrieving the non-frontmatter content.
+ */
+export const extractFrontmatter = (content: string): FrontmatterExtractionResult => {
+  const lines = content.split('\n')
+  if (lines.length < 2 || lines[0].length < 3 || lines[0][0] !== '-' || !allCharsEqual(lines[0])) {
```

Comment:
I'm confused. I can understand the first conditon. But why are you adding a new util function that does the same job as `lines[0] === '---'` ?

Reply:
It doesn't do the same job as you've written.
The `allCharsEqual` method - as stated in it's docs - checks that the given string only consists of the same letters. This condition is there to ensure that `---` is a valid line, but `---test` isn't valid.

Reply:
Is `----------` also valid as start line?

Reply:
Just out of curiosity: why don't you just a regex like `^-{3,}$`  ?